test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -20,15 +23,27 @@ public class PluginDiscovererTests
         [InlineData(" ")]
         public void Constructor_AcceptsAnyString(string rawPluginPaths)
         {
-            using (new PluginDiscoverer(rawPluginPaths))
+            var environmentalVariableReader = new Mock<IEnvironmentVariableReader>();
+            environmentalVariableReader.Setup(env => env.GetEnvironmentVariable(EnvironmentVariableConstants.DesktopPluginPaths)).Returns(rawPluginPaths);
+            environmentalVariableReader.Setup(env => env.GetEnvironmentVariable(EnvironmentVariableConstants.CorePluginPaths)).Returns(rawPluginPaths);
+
+            Exception exception = Record.Exception(() =>
             {
-            }
+                using (new PluginDiscoverer())
+                {
+                }
+            });
+
+            Assert.Null(exception);
```

Comment:
Why not simply?

```C#
   using (new PluginDiscoverer())
   {
   }
```

An exception would fail the test.

Reply:
https://github.com/NuGet/NuGet.Client/pull/5990#discussion_r1809632533